packages/thumbprint-react/components/Tooltip/index.tsx

```diff
@@ -102,10 +111,13 @@ export default function Tooltip({
     text,
     children,
     closeDelayLength = 200,
+    persistTooltipOnClick,
+    ariaLabel,
 }: TooltipProps): JSX.Element {
     const [isOpen, setIsOpen] = useState<boolean>(false);
     const [openTimeout, setOpenTimeout] = useState<number | undefined>(undefined);
     const [closeTimeout, setCloseTimeout] = useState<number | undefined>(undefined);
+    const [hasOpenedByOnClick, setHasOpenedByOnClick] = useState<boolean>(false);
```

Comment:
Rather than making a separate `useState`, I think it'd be better to modify the `isOpen` state to be an object rather than a boolean:

```typescript
const [isOpen, setIsOpen] = useState<
    {
        isOpen: true,
        reason: 'click' | 'hover'
    } | {
        isOpen: false,
    }
>(false);
```

This guarantees that we avoid introducing a bug in the code where we're in an impossible state such as `isOpen = false` and `hasOpenedByOnClick = true`. It should also simplifies the code a bit since you won't have to "clean up" the value of `hasOpenedByOnClick` when `hide` or `setIsOpen`.

Reply:
That's a good suggestion! Updated!